### PR TITLE
AF-1528: Turn off M2ServletContextListener by default and some logging improvements

### PR DESCRIPTION
**Pull request by porcelli**
This PR turns off M2ServletContextListener if tooling is not running in offline mode, so avoiding touching local maven repo.
This aims to bring more stability for kie web developers, avoiding potential override of their local maven repo.

### CODE REVIEW

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java [hunk 1]
```diff
@@ -88,6 +97,32 @@ public void deployJarsFrowWar() throws Exception {
...
+    @Test
+    public void checkRegularExecution() {
+        System.setProperty("kie.maven.offline.force", "false");
+        final M2ServletContextListener listener = spy(new M2ServletContextListener());
+        final ServletContextEvent contextEvent = mock(ServletContextEvent.class);
+        final ServletContext servletContext = mock(ServletContext.class);
+        when(servletContext.getRealPath(any())).thenReturn(".");
+        when(contextEvent.getServletContext()).thenReturn(servletContext);
+        listener.contextInitialized(contextEvent);
+        System.clearProperty("kie.maven.offline.force");
```
Comment (jhrcek) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java]: Please move this `clearProperty` call to `@After` to guarantee cleanup even if the test fails.

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java [hunk 2]
```diff
@@ -149,16 +153,17 @@ private Properties readZipFile(String zipFilePath) {
...
 
-    public void deploy(final GAV gav,
-                       final String jarPath) {
+    private void deploy(final GAV gav,
+                        final String jarPath) {
 
         final Artifact artifact = MavenRepository.getMavenRepository().resolveArtifact(new AFReleaseIdImpl(gav.getGroupId(), gav.getArtifactId(), gav.getVersion()));
 
         if (artifact != null) {
+            logger.info("Maven Artifact {} already exists.", gav.toString());
```
Comment (jhrcek) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: Please make this log message a bit more informative. Where does it exist (so user knows where he can delete it if needed)?

**Comment by manstis** [comment]
@jhrcek Please see the "Workbench webapps fail to run" email thread on BSIG-ALL.

**Comment by porcelli** [comment]
@jhrcek this is not exactly a fix (as the issue pointed by @manstis I couldn't reproduce neither), it's more an alginment that you'd just need this turned on if you're offline, otherwise maven infra itself will take care of any missing dependency.

**Comment by manstis** [comment]
@porcelli This is good to merge.

**Pull request merged**